subt_ign/src/GameLogicPlugin.cc

```diff
@@ -777,6 +786,9 @@ void GameLogicPlugin::PostUpdate(
               std::chrono::duration_cast<std::chrono::steady_clock::duration>
               (std::chrono::seconds(s) + std::chrono::nanoseconds(ns));
 
+          // check if the robot has flipped
+          this->dataPtr->CheckRobotFlip(name, pose);
```

Comment:
Should we store whether the robot is already flipped? Otherwise, once it flips, the log file will be flooded with flip events. It might also be a good idea to record an event only after the robot stays flipped for a duration of time to avoid false positives.

Reply:
I added a data structure (map) that keeps track of the robot's flip information in [40f1186](https://github.com/osrf/subt/pull/519/commits/40f1186627e45ad30f27ed98f358de4c92d17b67). I used this map to make sure a robot has been flipped for a few seconds before logging a flip event, and also used it to only log a flip event once per new flip.

Reply:
This works for me.